python/xorbits/_mars/tensor/fuse/jax.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+try:
+    import jax
+
+    JAX_INSTALLED = True
+except ImportError:
+    JAX_INSTALLED = False
+
+from .. import arithmetic, reduction
+from ..array_utils import as_same_device
+from ..operands import TensorFuse
+from .core import TensorFuseChunkMixin
+
+
+class TensorJAXFuseChunk(TensorFuse, TensorFuseChunkMixin):
+    _op_type_ = None  # no opcode, cannot be serialized
+
+    @classmethod
+    def execute(cls, ctx, op):
+        chunk = op.outputs[0]
+        inputs = as_same_device([ctx[c.key] for c in op.inputs], device=op.device)
```

Comment:
Now the inputs should have already been on the same device before execution. This is guaranteed by the fetch mechanism.
```suggestion
        inputs = [ctx[c.key] for c in op.inputs]
```